pkg/kn/flags/podspec_helper.go

```diff
@@ -378,6 +380,26 @@ func UpdateImagePullPolicy(spec *corev1.PodSpec, imagePullPolicy string) error {
 	return nil
 }
 
+func UpdateDefaultSecurityContext(spec *corev1.PodSpec) {
+	container := containerOfPodSpec(spec)
+	if container.SecurityContext == nil {
+		container.SecurityContext = DefaultSecCon()
+	}
+}
+
+func DefaultSecCon() *corev1.SecurityContext {
+	return &corev1.SecurityContext{
+		AllowPrivilegeEscalation: pointer.Bool(false),
+		RunAsNonRoot:             pointer.Bool(true),
```

Comment:
My question would be if this does not break the existing setup and whether this is not too restrictive. I mean, there are still tons of container images that can only run as root and/or need certain capabilities. Having such a strict default (that currently can't be changed), is quite drastic. I think we should only add this if there is at least an option to get back the previous behaviour, i.e. having no `SecurityContext` at all. This then could be documented as a change (which is still not backwards compatible). This option could be called `--no-security-context` or maybe even better something like `--security-context=none` (with the alternatives like `--security-context=strict` (the default) and `--security-context=privilegeEscalation:allow,runAsNonRoot:true,dropCapabilities:NET` for fine grained security context configuration (this we can add later but `strict` and `none` should be already along with this PR.

Reply:
That's at least ready with on/off switch. https://github.com/knative/client/pull/1821/commits/0fcf8000185bdf24ed98893d8b103b494e846ade